### PR TITLE
Percentage of winning

### PR DESCRIPTION
**Pull request by BONSURHA**
in this branch I added the following features:
- new POST function: `/get_winning_percentage`
- new input board type to the LLM, matrix format instead of the complete FEN
- add the color of the user in the input to the LLM

Fix the following issue:
- non JSON now provide a specific error message, before was not handle in a correct way
- update the `api.md` for the frontend team
- update the backend `README.md`, the guide is more clear

### CODE REVIEW

**Comment by BONSURHA** [comment]
@thorsjur I added you so you can see the API end-point in order to update in the front-end when you have time

**Comment by thorsjur** [comment]
> @thorsjur I added you so you can see the API end-point in order to update in the front-end when you have time

I think @Thalamicks has already implemented the frontend part in the his PR.

**Comment by BONSURHA** [comment]
Sorry @thorsjur, I forgot to re move you from this PR because I “collaborated” with Federico in his branch. My bad

File: backend/app/LLM_engine/main_coach.py [hunk 1]
```diff
@@ -10,62 +10,76 @@ def __init__(self, api_key):
...
         self.reset_automatic_history()
         self.prompt = """You are a chess coach. Your role is to advise and coach the user on playing chess.
-                For each move, you will evaluate the user’s move and provide feedback on it, using the Stockfish engine to guide your assessments.
-                Additionally, and this is the most challenging part, you should be ready to answer any questions the user has about the current game state,
-                the move you suggested, move sequences, or any general advice they may seek."""
-        # self.automatic_conversation.append({"role": "user", "content": self.prompt})
+                            For each move, you will evaluate the user’s move and provide feedback on it, using the Stockfish engine to guide your assessments.
+                            Additionally, and this is the most challenging part, you should be ready to answer any questions the user has about the current game state,
+                            the move you suggested, move sequences, or any general advice they may seek."""
+        self.automatic_conversation.append({"role": "user", "content": self.prompt})
```
Comment (Gjovani872) [inline backend/app/LLM_engine/main_coach.py:16]: why did we add this back ?

File: backend/app/LLM_engine/utils/utils.py [hunk 2]
```diff
@@ -7,3 +8,27 @@ def get_llm_api():
...
+def get_player(fen_player):
+    if fen_player == 'w':
+        return 'white'
+    else:
+        return 'black'
+
+def from_fen_to_board(fen):
+    board = get_string_board(fen)
+    matrixboard = [[' ' for _ in range(8)] for _ in range(8)]
+    # matrixboard = np.zeros((8,8))
```
Comment (Gjovani872) [inline backend/app/LLM_engine/utils/utils.py:24]: remove zombie code

**Review by Gjovani872 — APPROVED** [review]
Looks good to me!

File: backend/app/LLM_engine/main_coach.py (hunk 1 above)
Comment (BONSURHA) [reply backend/app/LLM_engine/main_coach.py:16]: mistral work with this

**Pull request merged**